### PR TITLE
Specify attribute types of SyntaxError

### PR DESCRIPTION
**Pull request by dahlia**
`SyntaxError` has its own special attributes.  This patch specifies them and their types.

### CODE REVIEW

**Comment by dahlia** [comment]
Fixed the failed build.

**Comment by gvanrossum** [comment]
It looks to me like the test failure is an anomaly. I'm going to merge this. Thanks for the PR!

File: stdlib/2.7/__builtin__.pyi [hunk 1]
```diff
@@ -814,7 +814,11 @@ class OSError(EnvironmentError): ...
 class WindowsError(OSError): ...
 class OverflowError(ArithmeticError): ...
 class ReferenceError(StandardError): ...
-class SyntaxError(StandardError): ...
+class SyntaxError(StandardError):
+    msg = ...  # type: str
+    lineno = ...  # type: Union[int, long]
+    offset = ...  # type: Union[int, long]
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: Can you make these plain int? We don't really care about the difference between int and long, and mypy treats them the same. There are (almost) no other stubs mentioning the long type.

Comment (dahlia) [inline stdlib/2.7/__builtin__.pyi]: Okay, I will change this.

**Comment by dahlia** [comment]
Amended!

**Comment by gvanrossum** [comment]
Thanks! I am going to merge even though the test fails.

**Pull request merged**